### PR TITLE
Updated unit tests in src/transport/tests/ to use PW instead of NL.

### PR DESCRIPTION
**Pull request by feasel0**
These changes are replacing NL unit tests with pigweed unit tests. I updated the tests in src/transport/tests/

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/transport/tests/TestSecureSessionTable.cpp [hunk 1]
```diff
@@ -131,10 +128,25 @@ void TestSecureSessionTable::CreateSessionTable(std::vector<SessionParameters> &
     }
 }
 
-void TestSecureSessionTable::ValidateSessionSorting(nlTestSuite * inSuite, void * inContext)
+class SecureSessionTableTest : public ::testing::Test
```
Comment (jlatusek) [inline src/transport/tests/TestSecureSessionTable.cpp:131]: Merge with `TestSecureSessionTable`

File: src/transport/tests/TestSessionManager.cpp [hunk 2]
```diff
@@ -106,51 +103,53 @@ class TestSessMgrCallback : public SessionMessageDelegate
...
         ReceiveHandlerCallCount++;
     }
 
-    nlTestSuite * mSuite        = nullptr;
     int ReceiveHandlerCallCount = 0;
     bool LargeMessageSent       = false;
 };
 
-void CheckSimpleInitTest(nlTestSuite * inSuite, void * inContext)
+class TestSessionManager : public ::testing::Test
```
Comment (jlatusek) [inline src/transport/tests/TestSessionManager.cpp:120]: Merge it with `TestContext`

File: src/transport/tests/TestSessionManager.cpp [hunk 3]
```diff
@@ -979,31 +945,29 @@ static void SessionShiftingTest(nlTestSuite * inSuite, void * inContext)
...
     // Now shift the 1st session to the 3rd one, after shifting, holders should be:
     // 1. alice -> bob: stickyAliceToBobSession
     // 2. alice <- bob: bobToAliceSession
     // 3. alice -> bob: aliceToBobSession, newAliceToBobSession
     sessionManager.GetSecureSessions().NewerSessionAvailable(newAliceToBobSession.Get().Value()->AsSecureSession());
 
-    NL_TEST_ASSERT(inSuite, aliceToBobSession);
-    NL_TEST_ASSERT(inSuite, stickyAliceToBobSession);
-    NL_TEST_ASSERT(inSuite, newAliceToBobSession);
+    EXPECT_TRUE(!!aliceToBobSession);
```
[the remark this answers is not in the document]
Comment (andy31415) [reply src/transport/tests/TestSessionManager.cpp:957]: It seems we only have operator bool on these. I guess we could do static_cast<bool> or Get().HasValue() however both looked ugly as well. the `!!` does feel javascript-like however I had no good suggestions. Do you have any?

[the remark this answers is not in the document]
Comment (jlatusek) [reply src/transport/tests/TestSessionManager.cpp:957]: I've tried it locally, and the compiler doesn't have a problem with automatic conversion to a bool value in the `EXPECT_TRUE` expression. In [doc](https://en.cppreference.com/w/cpp/language/implicit_conversion), I found out that `Contextual conversions` should be made for `the controlling expression of if, while, for;`

[the remark this answers is not in the document]
Comment (feasel0) [reply src/transport/tests/TestSessionManager.cpp]: Ok this seems to work.  Removing `!!`.

File: src/transport/tests/TestSessionManager.cpp [hunk 4]
```diff
@@ -137,8 +134,8 @@ TEST_F(TestSessionManager, CheckSimpleInitTest)
 
     EXPECT_EQ(CHIP_NO_ERROR, fabricTableHolder.Init());
     EXPECT_EQ(CHIP_NO_ERROR,
-              sessionManager.Init(&inContext->GetSystemLayer(), &inContext->GetTransportMgr(), &gMessageCounterManager,
-                                  &deviceStorage, &fabricTableHolder.GetFabricTable(), sessionKeystore));
+              sessionManager.Init(&this->GetSystemLayer(), &this->GetTransportMgr(), &gMessageCounterManager, &deviceStorage,
```
Comment (jlatusek) [inline src/transport/tests/TestSessionManager.cpp:137]: You can just call functions, you don't have to use `&this->`

**Review by jlatusek — CHANGES_REQUESTED** [review]

File: src/transport/tests/TestSessionManager.cpp (hunk 4 above)
Comment (feasel0) [reply src/transport/tests/TestSessionManager.cpp]: Updated in all places.

File: src/controller/tests/data_model/BUILD.gn [hunk 5]
```diff
@@ -40,4 +40,4 @@ chip_test_suite_using_nltest("data_model") {
     "${chip_root}/src/transport/raw/tests:helpers",
     "${nlunit_test_root}:nlunit-test",
   ]
-}
+}
```
Comment (andy31415) [inline src/controller/tests/data_model/BUILD.gn:43]: some of these files seem to have lost the trailing newline. Could we add it back?

**Pull request closed without merging**